README.md

```diff
@@ -236,6 +236,34 @@ Pass the `--analyze` option to have Webpack generate a `stats.json` file for the
 aegir build --analyze
 ```
 
+### Generating types
+
+This will generate `.d.ts` files from files in your project.  Suitable for use in a `prepublishOnly` [npm script](https://docs.npmjs.com/misc/scripts).
+
+```console
+$ aegir generate-types PATTERN
+```
+
+It supports a glob pattern as the argument, but to prevent cli expansion you should wrap it in quotes:
+
+```console
+$ aegir generate-types 'src/**/*.js'
+```
+
+Pass the `--overwrite` option to remove `.d.ts` files that would be generated:
+
+```console
+$ aegir generate-types --overwrite 'src/**/*.js'
+```
+
+Any flags or config to pass to `tsc` can be specified as forward options:
+
+```console
+$ aegir generate-types --overwrite 'src/**/*.js' -- --allowJs --esModuleInterop
```

Comment:
Please note that `tsconfig.json` affects a lot what options can be passed or how they collide. E.g. having `noEmit` compiler option will prevent `emitDeclarationOnly` passed here.

I don't think documenting every single collision is realistic, but maybe a note or a link to an example setup might be helpful.